### PR TITLE
Cutfem final changes (3D)

### PR DESCRIPTION
**Pull request by cekees**
# Mandatory Checklist

Please ensure that the following criteria are met:

- [x] Title of pull request describes the changes/features
- [x] Request at least 2 reviewers
- [x] If new files are being added, the files are no larger than 100kB. Post the file sizes. EQUIV. POLY. KERNAL file is/will remain larger that 100kb.
- [x] Code coverage did not decrease. If this is a bug fix, a test should cover that bug fix. If a new feature is added, a test should be made to cover that feature.
- [ ] New features have appropriate documentation strings (readable by sphinx)
- [x] Contributor has read and agreed with CONTRIBUTING.md and has added themselves to CONTRIBUTORS.md 

As a general rule of thumb, try to follow [PEP8](https://www.python.org/dev/peps/pep-0008/) guidelines.

# Description

### CODE REVIEW

File: proteus/equivalent_polynomials_coefficients_quad.h [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+                                double* b_H, double* b_ImH, double* b_D)
+  {
+    b_H[0] = (1.0/6.0)*(-4*std::pow(theta02, 4)*std::pow(theta31, 2) + 8*std::pow(theta02, 4)*theta31 - 4*std::pow(theta02, 4) + 2*std::pow(theta02, 3)*std::pow(theta31, 2) - 4*std::pow(theta02, 3)*theta31 + 2*std::pow(theta02, 3)*std::sqrt((2*std::pow(theta02, 2)*std::pow(theta31, 2) - 4*std::pow(theta02, 2)*theta31 + 2*std::pow(theta02, 2) + std::pow(theta31, 2) - 2*theta31 + 1)/(std::pow(theta31, 2) - 2*theta31 + 1))*std::sqrt(2*std::pow(theta02, 2) + 1)*(std::pow(theta31, 2) - 2*theta31 + 1) + 2*std::pow(theta02, 3) - 4*std::pow(theta02, 2)*std::pow(theta31, 2) + 8*std::pow(theta02, 2)*theta31 - 4*std::pow(theta02, 2) + theta02*std::pow(theta31, 2) - 2*theta02*theta31 + theta02 - std::pow(theta31, 2) + 2*theta31 + (theta02*(theta01*theta31 - theta01 - theta31 + 1) + theta31*(theta02*theta32 - theta02 - theta32 + 1))*(4*std::pow(theta02, 4)*theta31 - 4*std::pow(theta02, 4) + 4*std::pow(theta02, 2)*theta31 - 4*std::pow(theta02, 2) + theta31 - 1) - 1)/(4*std::pow(theta02, 4)*theta31 - 4*std::pow(theta02, 4) + 4*std::pow(theta02, 2)*theta31 - 4*std::pow(theta02, 2) + theta31 - 1);
```
Comment (zhang-alvin) [inline proteus/equivalent_polynomials_coefficients_quad.h:16]: where does this monstrous formula come from?

File: proteus/mesh.pxd [hunk 2]
```diff
@@ -155,6 +155,7 @@ cdef extern from "mesh.h":
                                                                     const double& Ly,
                                                                     const double& Lz,
                                                                     Mesh& mesh)
+    cdef int reorientTetrahedralMesh(Mesh& mesh)
```
Comment (zhang-alvin) [inline proteus/mesh.pxd:158]: could you briefly describe what reorient means here?

File: proteus/equivalent_polynomials_coefficients_quad.h (hunk 1 above)
Comment (cekees) [reply proteus/equivalent_polynomials_coefficients_quad.h:16]: It's generated from scripts/gen_eqp_quad.py. These are the integrals the polynomials basis functions time the Heaviside function (or equivalently the integrals of the polynomials on cut cells).

File: proteus/mesh.pxd (hunk 2 above)
Comment (cekees) [reply proteus/mesh.pxd:158]: it checks the sign of the Jacobian determinant for each tet and swaps the order of two nodes in the elementNodesArray if the determinant is negative. Most mesh generators already do that, but I found that the simple 3D tet from regular grid algorithm was not. Proteus generally doesn't care, but Paraview uses it to decide between "front faces" and "back faces".

**Comment by cekees** [comment]
@ejtovar at some point the gmsh installed via conda quit install properly on os x, which is why in this PR you see I've added `pip install gmsh` as a work around. Don't know what's going on, but you it's not an issue with this PR--our nightly build is also failing now as you can see from the "build failing" badge on the README.

**Pull request merged**